guildwars2/wallet.py

```diff
@@ -131,60 +151,43 @@ class WalletMixin:
         except discord.Forbidden:
             await ctx.send("Need permission to embed links")
 
-    @wallet.command(name="tokens")
+    @wallet.command(name="show", aliases=["keys", "maps", "tokens"])
     @commands.cooldown(1, 5, BucketType.user)
-    async def wallet_tokens(self, ctx):
-        """Shows instance-specific currencies
+    async def wallet_show(self, ctx):
+        """Shows currencies in your wallet
 
         Required permissions: wallet
         """
         try:
             doc = await self.fetch_key(ctx.author, ["wallet"])
-            results = await self.call_api("account/wallet", key=doc["key"])
         except APIError as e:
             return await self.error_handler(ctx, e)
-        ids = [5, 6, 9, 10, 11, 12, 13, 14, 7, 24, 28, 39]
-        embed = discord.Embed(
-            description="Tokens", colour=await self.get_embed_color(ctx))
-        for currency in results:
-            if currency["id"] not in ids:
-                continue
-            c_doc = await self.db.currencies.find_one({"_id": currency["id"]})
-            if c_doc["name"] == "Magnetite Shard":
-                embed.add_field(
-                    name=c_doc["name"], value=currency["value"], inline=False)
-            else:
-                embed.add_field(name=c_doc["name"], value=currency["value"])
-        embed.set_author(
-            name=doc["account_name"], icon_url=ctx.author.avatar_url)
-        embed.set_footer(
-            text=self.bot.user.name, icon_url=self.bot.user.avatar_url)
-        try:
-            await ctx.send(embed=embed)
-        except discord.Forbidden:
-            await ctx.send("Need permission to embed links")
+        await ctx.trigger_typing()
+        ids_cur = [1, 4, 2, 3, 18, 23, 15, 16, 50, 47]
+        ids_keys = [38, 37, 43, 41, 42, 40, 44, 49]
+        ids_maps = [25, 19, 27, 22, 20, 32, 45, 34]
+        ids_token = [5, 6, 9, 10, 11, 12, 13, 14, 7, 24]
+        ids_raid = [28, 39]
+        ids_l2 = [79280, 79899, 79469, 80332, 81127, 81706]
+        ids_l3 = [86069, 88955, 86977, 89537, 87645, 90783]
+        cur = await self.get_wallet(ctx, ids_cur)
+        keys = await self.get_wallet(ctx, ids_keys)
+        maps = await self.get_wallet(ctx, ids_maps)
+        token = await self.get_wallet(ctx, ids_token)
+        ls3 = await self.get_ls_currency(ctx, ids_l2)
+        ls4 = await self.get_ls_currency(ctx, ids_l3)
+        raid = await self.get_wallet(ctx, ids_raid)
```

Comment:
That's so many unnecessary API calls. Calling 5 times for wallet and 2 times for account search.

Reply:
- Changed get_wallet() and get_ls_currency() to accept a list of lists of ids and call API only once